CleverTapSDK/EventDatabase/CTEventDatabase.m

```diff
@@ -334,4 +341,67 @@ - (BOOL)deleteTable {
     return success;
 }
 
+- (BOOL)deleteLeastRecentlyUsedRows:(NSInteger)maxRowLimit
+              numberOfRowsToCleanup:(NSInteger)numberOfRowsToCleanup {
+    __block BOOL success = NO;
+    
+    dispatch_sync(_databaseQueue, ^{
+        // Begin a transaction to ensure atomicity
+        sqlite3_exec(_eventDatabase, "BEGIN TRANSACTION;", NULL, NULL, NULL);
+        
+        // Create an index on the `lastTs` column if it doesn't exist which will improve performance
+        // while deletion when table is large
+        const char *createIndexSQL = "CREATE INDEX IF NOT EXISTS idx_lastTs ON CTUserEventLogs(lastTs);";
+        char *errMsg = NULL;
+        int indexResult = sqlite3_exec(_eventDatabase, createIndexSQL, NULL, NULL, &errMsg);
+        
+        if (indexResult != SQLITE_OK) {
+            CleverTapLogInternal(self.config.logLevel, @"%@ Failed to create index on lastTs: %s", self, errMsg);
+            sqlite3_free(errMsg);
+            sqlite3_exec(_eventDatabase, "ROLLBACK;", NULL, NULL, NULL);  // Rollback transaction if index creation fails
+            return;
+        }
+        
+        NSString *countQuerySQL = @"SELECT COUNT(*) FROM CTUserEventLogs;";
+        sqlite3_stmt *countStatement;
+        if (sqlite3_prepare_v2(_eventDatabase, [countQuerySQL UTF8String], -1, &countStatement, NULL) == SQLITE_OK) {
+            if (sqlite3_step(countStatement) == SQLITE_ROW) {
+                NSInteger currentRowCount = sqlite3_column_int(countStatement, 0);
+                
+                // Calculate the number of rows to delete
+                NSInteger rowsToDelete = currentRowCount - (maxRowLimit - numberOfRowsToCleanup);
+                
+                if (rowsToDelete > 0) {
```

Comment:
maybe we can stop execution early if `currentRowCount` is less than `maxRowLimit`. 
Something like this: 
```
if (currentRowCount <= maxRowLimit) {
    sqlite3_exec(_eventDatabase, "ROLLBACK;", NULL, NULL, NULL);
    return YES;
}